src/pages/ViewVideo/ViewVideo.page.jsx

```diff
@@ -1,20 +1,24 @@
 import React from 'react';
 import { useParams } from 'react-router';
 import { VideoPlayer, ThumbnailCardList } from '../../components';
+import useYoutubeAPI from '../../utils/hooks/useYoutubeAPI';
 import { Grid } from './styled';
 
 function ViewVideoPage() {
   const { videoId } = useParams();
-  const allVideosData = JSON.parse(localStorage.getItem('videoData'));
-  const videoData = allVideosData.items.filter((obj) => {
-    return obj.id.videoId === videoId;
-  });
+  const videoData = useYoutubeAPI(null, videoId);
```

Comment:
You can see here that perhaps `useYoutubeAPI` has more than one responsibility. You don't want to run a search, but you want the videoData. Consider splitting the logic of `useYoutubeAPI` to have a hook for the `search` and a hook for the `videoId`. It will be easier to test it out and will simplify the need for running a condition on the `useEffect` (see comment above).